apps/system/js/app_usage_metrics.js

```diff
@@ -761,18 +814,21 @@
     // in the new batch and merge the new usage data with the old
     // usage data.
     for (var app in newbatch.data.apps) {
-      var newusage = newbatch.data.apps[app];
-      var oldusage = this.getAppUsage(app);
-
-      oldusage.usageTime += newusage.usageTime;
-      oldusage.invocations += newusage.invocations;
-      oldusage.installs += newusage.installs;
-      oldusage.uninstalls += newusage.uninstalls;
-
-      for (var url in newusage.activities) {
-        var newcount = newusage.activities[url];
-        var oldcount = oldusage.activities[url] || 0;
-        oldusage.activities[url] = oldcount + newcount;
+      var newdays = newbatch.data.apps[app];
+      for (var day in newdays) {
+        var newusage = newdays[day];
+        var oldusage = this.getAppUsage(app, day);
```

Comment:
I think day is a timestamp here, like '20150112', right? But getAppUsage() expects a Date object as the second argument, not a timestamp.  If I'm right it means that there are not tests for this case, or that you haven't run them yet.